library/sha256.c

```diff
@@ -37,6 +37,33 @@
 #if defined(__aarch64__)
 #  if defined(MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT) || \
     defined(MBEDTLS_SHA256_USE_A64_CRYPTO_ONLY)
+/* *INDENT-OFF* */
+#    if !defined(__ARM_FEATURE_CRYPTO)
+#      if defined(__clang__)
+#        if __clang_major__ < 18
+           /* TODO: Re-consider above after https://reviews.llvm.org/D131064
+            *       merged.
+            *
+            * The intrinsic declaration are guarded with ACLE predefined macros
+            * in clang, and those macros are only enabled with command line.
+            * Define the macros can enable those declaration and avoid compile
+            * error on it.
+            */
+#          define __ARM_FEATURE_CRYPTO 1
+#        endif
+#        pragma clang attribute push (__attribute__((target("crypto"))), apply_to=function)
+#        define MBEDTLS_POP_TARGET_PRAGMA
+#      elif defined(__GNUC__)
+#        if __GNUC__ < 6 /* TODO: check sha256 compatible for GCC */
```

Comment:
[change-request] 
```suggestion
#        if __GNUC__ < 5
```
according to https://gcc.gnu.org/gcc-5/changes.html . Earlier versions can use the crypto extensions when specifying a base `arch` or `cpu` that has them, though, so erroring out on GCC &lt; 5 isn't always correct either. But GCC 5 was released in 2015, so at this point it's fine that we don't try hard to support earlier versions.

Or maybe we should omit this check, and let `#pragma GCC target` error out if GCC is too old?

Reply:
I think this change will help user. 
Although GCC-5 announce it, but some functions are missed, so I remove gcc-5 support.

Those miss function can be workaround. Before we workaround them, I do not think we should say it is supported

Reply:
Add some comment about gcc-5